src/User/Service/Currency/UpdateCurrencyService.php

```diff
@@ -41,27 +44,62 @@ public function __construct(
      */
     public function execute(int $userId, int $type, int $amount): void

```

Comment:
Please try to split these function into smaller parts, to get some cleaner code and not that much clustered statements.